set/ordered.go

```diff
@@ -27,3 +27,15 @@ func (u *OrderedSet[T]) Contains(val T) bool {
 func (u *OrderedSet[T]) Values() []T {
 	return u.values
 }
+
+func (u *OrderedSet[T]) Size() int {
+	return len(u.valuesSet)
+}
+
+func (u *OrderedSet[T]) ValueAt(idx int, defaultValue T) T {
```

Comment:
Why do you set default value? 
This complicates the usage as the client code will not know if the value returned is default or not. 
Just use index and let client code to check length. This function should be single responsible.